Allow this script to be used in a team

ram535ii opened the pull request:
You can now allow every user in a team to be notified individually that
they have been assigned to a PR. Make sure that when listing the Github
usernames and the Rooms to post in that they are both ordered
identically.

ram535ii committed:
Allow this script to be used in a team

You can now allow every user in a team to be notified individually that
they have been assigned to a PR. Make sure that when listing the Github
usernames and the Rooms to post in that they are both ordered
identically.

ram535ii commented:
@seddy Not to keep hassling you, but I have generalized the script so that it would work for a team. Functionally very little has changed, but any tips would be useful. Especially with regards to the setup explanation at the top and how I have pulled out the sanitization method.

seddy commented on scripts/gh-pr-assigned.coffee:
```diff
@@ -1,9 +1,14 @@
+#   HUBOT_GITHUB_USERNAMES - The github user to check for pull request assignment
+#   (NOTE: rooms and usernames should match in order
+#   i.e. HUBOT_GITHUB_EVENT_NOTIFIER_ROOMS = room1, room2
+#        HUBOT_GITHUB_USERNAMES            = user1, user2
```
Not sure how scalable this is:

```
HUBOT_GITHUB_EVENT_NOTIFIER_ROOMS = room1, room2, chat4, totallyawesomeplace, slackisdabomb, fullyfunctionalstuff, js-data-awesomeness, <insert 20 more rooms here>
HUBOT_GITHUB_USERNAMES            = user1, user2, bob, janice, ross, chandler, rachel, phoebe, monica, joey, <insert X more people here>
```

So given I've configured the above, I'm Phoebe and I want to change my room.... wait, which room am I notifying to?

It might be better to do this as tuplets:

```
HUBOT_GITHUB_ROOM_MAPPINGS = user1|room1, user2|room2, joey|sandwichhouse, etc etc
```

And modify `splitAndSanitiseEnvVars` accordingly:

``` coffee
# This will return an array of tuples, so:
#   [[user1, room1], [user2, room2], ...]
splitAndSanitiseEnvVars = (commaSeparatedList) ->
  commaSeparatedList.split(",").map((mapping) ->
    mapping.split("|") -> ((allocation) ->
      [allocation[0], allocation[1]]
    )
  )
```

Does that make sense?

ram535ii committed:
Move from two env variables to 1

The way the mapping of username>room is much easier and way more
scaleable for multiple users.
